src/core/dep-graph.ts

```diff
@@ -138,6 +138,38 @@ class DepGraphImpl implements types.DepGraphInternal {
     return count;
   }
 
+  public isSame(other: types.DepGraph): boolean {
+    const sortByNodeId = (nodeA: any, nodeB: any) => nodeA.nodeId.localeCompare(nodeB.nodeId);
+
+    const selfNodes = this.toJSON().graph.nodes.sort(sortByNodeId);
+    const otherNodes = other.toJSON().graph.nodes.sort(sortByNodeId);
+
+    if (selfNodes.length !== otherNodes.length) {
+      return false;
+    }
+
+    for (let i = 0; i < selfNodes.length; i++) {
+      if (selfNodes[i].nodeId !== otherNodes[i].nodeId) {
```

Comment:
please compare not only the nodeIds, but also the data on each node, specifically the `pkgId` & `nodeInfo`

Reply:
Why? Can you explain please?

Reply:
The data the graph holds is:
* the relations between the *packages* (every node points to a `pkgId`)
* optional per node metadata
If one of these changes, than it's basically a different dep-graph

*actually* the `nodeIds` themselves don't matter at all, what matters is what `pkgId` they represent.

So the perfect comparison will ignore `nodeIds` and will compare the actual relations. This is not a must if this complicates the solution too much right now and blocks you, as usually the dep-graphs will be constructed in the same way so the `nodeId` of same graphs *should* be the same, but it's not always the case ...

Reply:
done